src/components/views/messages/TextualBody.js

```diff
@@ -37,28 +38,76 @@ module.exports = React.createClass({
 
         /* link URL for the highlights */
         highlightLink: React.PropTypes.string,
+
+        /* callback for when our widget has loaded */
+        onWidgetLoad: React.PropTypes.func,
+    },
+
+    getInitialState: function() {
+        return {
+            link: null,
```

Comment:
given `link` never changes, I'd like to see it set before the component is mounted, and probably saved in `this` rather than `state`. This will save a `render()` cycle, and simplify `shouldComponentUpdate`.